Tests/NetworkServiceTests/NetworkServiceTests.swift

```diff
@@ -1,8 +1,170 @@
-import NetworkService
 import XCTest
+import NetworkService
 
 final class NetworkServiceTests: XCTestCase {
-    func testUseMethodExistsInAPI() {
-        NetworkService().testUse()
+    
+    var networkService: NetworkService<MockEndpoint>!
+    
+    override func setUpWithError() throws {
+        networkService = NetworkService()
+        URLProtocol.registerClass(MockURLProtocol.self)
+    }
```

Comment:
Does it make sense to create a private variable `endpoint` not to create a new `endpoint` in every test case (see the suggestion on lines 169-170).
```suggestion
        URLProtocol.registerClass(MockURLProtocol.self)
        endpoint = MockEndpoint()
    }
```

Reply:
fixed with 036eb7f